### PR TITLE
feat: upgrade fundamental-styles to 0.11.0-rc.4

### LINKED ISSUE #1071: Popover: styling is incorrect on docs site
### Description

Not sure if this is from the way the examples are constructed or if we missed something on the last few styles upgrades. 

Screen Shot 2020-06-02 at 6 45 19 AM

### Versions
**fundamental-react:** 0.10.2-rc.0

---

_**NOTE:** Where applicable, please include uncropped screen captures._

_**DISCLAIMER:**
After triaging an issue, the fundamental-react team will see if it can be reproduced or confirmed. If more information is needed, the fundamental-react team will contact the author.  Any issues awaiting responses from the author for more than 7 days will be closed.  The author can re-open the issue at a later time if they can present the requested information._

### PR DESCRIPTION
**Pull request by jbadan**
### Description

BREAKING CHANGES:
- `Panel`: renamed `LayoutPanel`
- `Select`: underlying html changes
- `StepInput`: underlying html changes
- `Tile`: removed `active` prop
- `Tile`: `onClick` prop now required - all tiles are clickable by default
- `Tile`:  removed `productTile` prop
- `Tile`: removed `tabIndex` prop (no longer necessary because they are always buttons now)
- `Tile.Actions` removed
- `Tile.Media` removed
- `Tile.Content` no longer takes a `title` prop - this should now be added as a child to `Tile.Header`
- `Tile.Content`: removed `headingLevel` prop 
- `Tile.Content`: removed `titleProps` prop
- `Tile.Content`: removed `productTile` prop
- `Tile.Content`: underlying html changes related to removal of title 

Features:
- `List.Item`: add `selected` state
- `Select`: keyboard handling + focus handling
- `StepInput`: new `compact` state
- `Tile`: new `isDouble` prop (this is a bad name Help!) - maxes tile 2x1
- `Tile`: new `size` prop
- `Tile.Header`: new subcomponent
- `Tile.Footer`: new subcomponent
- `Tile.Content`: new `twoColumns` prop - enabled Content to be split into 2 columns

Fixes:
- fix Popover styling issues on docs site

fixes #1071

### CODE REVIEW

File: src/LayoutPanel/LayoutPanel.js [hunk 1]
```diff
@@ -48,11 +48,11 @@ Panel.propTypes = {
...
-Panel.Filters = PanelFilters;
-Panel.Footer = PanelFooter;
-Panel.Head = PanelHead;
-Panel.Header = PanelHeader;
+LayoutPanel.Actions = LayoutPanelActions;
+LayoutPanel.Body = LayoutPanelBody;
+LayoutPanel.Filters = LayoutPanelFilters;
+LayoutPanel.Footer = LayoutPanelFooter;
+LayoutPanel.Head = LayoutPanelHead;
+LayoutPanel.Header = LayoutPanelHeader;
```
Comment (jbadan) [inline src/LayoutPanel/LayoutPanel.js:56]: I've exported something wrong in layout panel but I can't figure out what 👀

File: src/Select/Select.test.js [hunk 2]
```diff
@@ -51,7 +51,7 @@ describe('<Select />', () => {
         });
     });
 
-    test('forwards the ref to the button', () => {
+    xtest('forwards the ref to the div role="button"', () => {
```
Comment (jbadan) [inline src/Select/Select.test.js:54]: @skvale I've done something wrong with how to handle forwarding the ref in Select - could you take a look when you get some time?

File: src/Select/Select.js [hunk 3]
```diff
@@ -26,18 +28,122 @@ const Select = React.forwardRef(({
 
     useEffect(() => {
         if (!disableStyles) {
+            require('fundamental-styles/dist/button.css');
+            require('fundamental-styles/dist/icon.css');
             require('fundamental-styles/dist/select.css');
         }
     }, []);
 
+    const divRef = ref ? ref : useRef(null);
```
Comment (skvale) [inline src/Select/Select.js:37]: We shouldn't conditionally use hooks
```js
const internalDivRef = useRef(null);
const divRef = ref ? ref : internalDivRef;
```

File: src/Select/Select.test.js (hunk 2 above)
Comment (skvale) [reply src/Select/Select.test.js:54]: Popover is overriding the ref prop of this control element
```js
        let controlProps = {
            onClick: onClickFunctions,
            ref: (c) => {
                this.controlRef = findDOMNode(c);
            }
        };
```
I'll see if there's way to extend the ref, but still maintain the original ref somehow

Comment (skvale) [reply src/Select/Select.test.js:54]: I think we need to do something like this in Popover
```js
let controlProps = {
    onClick: onClickFunctions,
    ref: (c) => {
        this.controlRef = findDOMNode(c);
        if (typeof control.ref === 'function') {
            control.ref(c);
        } else if (typeof control.ref === 'object') {
            control.ref.current = c;
        }
    }
};
```

File: src/StepInput/StepInput.js [hunk 4]
```diff
@@ -25,11 +36,19 @@ const StepInput = React.forwardRef(({
...
+
+    const stepInputClasses = classnames(
         className,
-        'fd-input-group--control',
+        'fd-step-input',
         {
+            'fd-step-input--compact': compact,
             'is-disabled': disabled,
+            'is-readonly': readOnly,
             [`is-${validationState?.state}`]: validationState ?.state
```
Comment (skvale) [inline src/StepInput/StepInput.js:52]: I wonder why eslint doesn't pick up on this line having an extra space `validationState ?.state`

File: src/Select/Select.js [hunk 5]
```diff
@@ -26,18 +28,124 @@ const Select = React.forwardRef(({
...
+            case 'tab':
+                e.stopPropagation();
+                setIsExpanded(false);
+                setFocusedElement(tryFocus(divRef.current));
+                break;
+            case 'enter':
+            case 'space':
+                e.stopPropagation();
+                handleSelect(e, option);
+                setFocusedElement(tryFocus(divRef.current));
```
Comment (skvale) [inline src/Select/Select.js:135]: It looks like the intent of the `enter` and `space` is to close the select.  This isn't happening.
For some reason it is calling the `handleClick` immediately after the `handleSelect` and the `handleClick` is toggling the `isExpanded` state back to true

Comment (skvale) [reply src/Select/Select.js:135]: It works as expected if you call `e.preventDefualt();` though

File: src/StepInput/StepInput.js (hunk 4 above)
Comment (jbadan) [reply src/StepInput/StepInput.js:52]: I'm not sure - good catch though!

File: src/Button/__stories__/__snapshots__/ButtonSegmented.visual.storyshot [hunk 6]
```diff
@@ -1,60 +1,5 @@
 // Jest Snapshot v1, https://goo.gl/fbAQLP
 
-exports[`Storyshots Visual Button Group 1`] = `
```
Comment (jbadan) [inline src/Button/__stories__/__snapshots__/ButtonSegmented.visual.storyshot:3]: this was a duplicate of the same storyshot below

**Pull request merged**